addons/godot-xr-tools/assets/PlayerBody.gd

```diff
@@ -39,16 +39,16 @@ export var push_rigid_bodies := true
 export (Resource) var physics = null setget set_physics, get_physics
 
 ## Path to the ARVROrigin node
-export (NodePath) var origin = null
+export var origin := NodePath()
 
 ## Path to the ARVRCamera node
-export (NodePath) var camera = null
+export var camera := NodePath()
 
 ## ARVROrigin node
-var origin_node: ARVROrigin = null
+onready var origin_node := ARVRHelpers.get_arvr_origin(self, origin)
```

Comment:
Just a warning for the future, I prefer onready var notation like this but Godot 4 has an interesting "bug" that may or may not get fixed. Having onready vars counts as having a `_ready` function even if you haven't added one to the script. If you have a `_ready` function you now have to call `super._ready()` or the parents ready function won't get called.

Doesn't impact Godot 3 but something to start keeping in mind for porting the code to our Godot 4 branch, which we should start doing so we don't deviate to far

Reply:
Ugh, annoying but good to know. Languages are supposed to get easier over time and end up achieving more with less typing while at the same time defaulting to correct behavior. This sounds like a regression from Godot 3 that's going to mess people up.

When the language has the 'onready' keyword it's already an acknowledgement that _ready() isn't a normal function. Hopefully they at least add a static analysis warning/error if the _ready does not chain.

Reply:
yeah i'm on the fence about the change too, but it does add in the ability to decide when the parent function is run, sometimes you want to be able to run logic before hand.